python-avd/pyavd/_eos_designs/ip_addressing/__init__.py

```diff
@@ -265,6 +351,9 @@ def vtep_ip_mlag(self) -> str:
         if self._vtep_loopback_ipv4_address:
             return self._vtep_loopback_ipv4_address
 
+        if not self.shared_utils.underlay_ipv4:
+            return None
```

Comment:
We should never call this function if we don't want the IP. Please adjust the calling logic instead.
Please note that this class is meant to be subclassed by users, so we cannot rely on adding special logic here. That same logic will not be in the user's variant.